operator/internal/controller/pulumi/git.go

```diff
@@ -325,7 +326,9 @@ func (sess *StackReconcilerSession) setupWorkspaceFromGitSource(ctx context.Cont
 		}
 	}
 
-	sess.ws.Spec.Git.Auth = auth
+	if !reflect.DeepEqual(*auth, autov1alpha1.GitAuth{}) {
+		sess.ws.Spec.Git.Auth = auth
+	}
```

Comment:
Is this needed? The v1 controller is using the struct's zero value to represent unauthenticated, which I think should still be OK here.

Reply:
I did it for ergonomic reasons while writing a test case.